docs/io/ascii/howtoexamples.rst

```diff
@@ -0,0 +1,209 @@
+a format close to, but not identical with, any of the `Supported formats`_,
+the fastest solution is often to open that one table file
+in a text editor to modify it until it does conform to a format that
+any of the readers above can already read. On the other hand, if we need to
+read table of that specific format again and again, it is better to find a
+to read them with `~astropy.io.ascii` without modifying every file by hand.
+
+
+Badly formatted header line
+---------------------------
+The following table will fail to parse (raising an
+`~astropy.io.InconsistentTableError') because the header line looks as
+if there were three columns, while in fact, there are only two::
+
+  Name spectral type
+  Vega A0
+  Altair A7
+
+Opening this file in a text editor to fix the format is easy::
+
+  Name "spectral type"
+  Vega A0
+  Altair A7
+
+or
+::
+
+  Name spectral_type
+  Vega A0
+  Altair A7
+
+would both work, e.g.
+
+..
+  EXAMPLE START
+  Make a table easier to read
+
+::
+
+  >>> from astropy.io import ascii
+  >>> table = """
+  ... Star "spectral type"
+  ... Vega A0
+  ... Altair A7
+  ... """
+  >>> ascii.read(table)
+  <Table length=2>
+   Star  spectral type
+   str6       str2
+  ------ -------------
+    Vega            A0
+  Altair            A7
+
+Similarly, the header could be commented out and the names of the table column
+can be set manually::
+
+  >>> from astropy.io import ascii
+  >>> table = """
+  ... #Star spectral type
+  ... Vega A0
+  ... Altair A7
+  ... """
+  >>> ascii.read(table, names=["Star", "spectral type"], format='no_header')
+  <Table length=2>
+   Star  spectral type
+   str6       str2
+  ------ -------------
+    Vega            A0
+  Altair            A7
+
+This last experiment shows us that reading works just fine, if we only
+find a way to ignore the badly formatted header line. That can be done without
+any modification of the table itself with the ``data_start`` parameter::
+
+   >>> table = """
+   ... Star spectral type
+   ... Vega A0
+   ... Altair A7
+   ... """
+   >>> ascii.read(table, names=["Star", "spectral type"], data_start=1)
+   <Table length=2>
+    Star  spectral type
+    str6       str2
+   ------ -------------
+     Vega            A0
+   Altair            A7
+
+..
+  EXAMPLE END
+
+Badly formatted data line
+-------------------------
+
+Similar principles apply to badly formatted data lines. Here is a
+table where the number of columns is not consistent (`alpha Cen`
+should be written as `"alpha Cen"` to make clear that the two words
+"alpha" and "Cen" are part of the same column)::
+
+  Star SpT
+  Vega A0
+  alpha Cen G2V+K1
+
+When we try to read that, astropy throws an
+`astropy.io.ascii.InconsistentTableError` and prints out a long list of formats
+that it tried on this table and some advice (some output omitted)::
+
+  >>> from astropy.io import ascii
+  >>> table = '''
+  ... Star SpT
+  ... Vega A0
+  ... alpha Cen G2V+K1
+  ... '''
+  >>> ascii.read(table)
```

Comment:
This is what I get in IPython or the Jupyter notebook without `guess=False` (copy and paste full output from the terminal. I think that's what most users use these days. We `guess=False`, I would not see "a long list of formats
that it tried on this table"

```
(kitchensink) MoritzAirRoseGold ~> ipython
Python 3.10.1 | packaged by conda-forge | (main, Dec 22 2021, 01:39:07) [Clang 11.1.0 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.30.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]:   >>> from astropy.io import ascii
   ...:   >>> table = '''
   ...:   ... Star SpT
   ...:   ... Vega A0
   ...:   ... alpha Cen G2V+K1
   ...:   ... '''
   ...:   >>> ascii.read(table)
---------------------------------------------------------------------------
InconsistentTableError                    Traceback (most recent call last)
~/mambaforge/envs/kitchensink/lib/python3.10/site-packages/astropy/io/ascii/ui.py in _guess(table, read_kwargs, format, fast_reader)
    521             reader = get_reader(**read_kwargs)
--> 522             dat = reader.read(table)
    523             _read_trace.append({'kwargs': copy.deepcopy(read_kwargs),

~/mambaforge/envs/kitchensink/lib/python3.10/site-packages/astropy/io/ascii/core.py in read(self, table)
   1368 
-> 1369                     raise InconsistentTableError(errmsg)
   1370 

InconsistentTableError: Number of header columns (2) inconsistent with data columns (3) at data line 1
Header values: ['Star', 'SpT']
Data values: ['alpha', 'Cen', 'G2V+K1']

During handling of the above exception, another exception occurred:

InconsistentTableError                    Traceback (most recent call last)
<ipython-input-1-732a26b21e0d> in <module>
      5 alpha Cen G2V+K1
      6 '''
----> 7 ascii.read(table)

~/mambaforge/envs/kitchensink/lib/python3.10/site-packages/astropy/io/ascii/ui.py in read(table, guess, **kwargs)
    339         # through below to the non-guess way so that any problems result in a
    340         # more useful traceback.
--> 341         dat = _guess(table, new_kwargs, format, fast_reader)
    342         if dat is None:
    343             guess = False

~/mambaforge/envs/kitchensink/lib/python3.10/site-packages/astropy/io/ascii/ui.py in _guess(table, read_kwargs, format, fast_reader)
    551                    '************************************************************************']
    552             lines.extend(msg)
--> 553             raise core.InconsistentTableError('\n'.join(lines))
    554 
    555 

InconsistentTableError: 
ERROR: Unable to guess table format with the guesses listed below:
Reader:Ecsv fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:FixedWidthTwoLine fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:RST fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:FastBasic fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Basic fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:FastRdb fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Rdb fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:FastTab fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Tab fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Cds fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Mrt fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Daophot fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:SExtractor fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Ipac fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:Latex fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:AASTex fast_reader: {'enable': True} fill_values: [('', '0')] strict_names: True
Reader:FastCommentedHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastCommentedHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastCommentedHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastCommentedHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastCommentedHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastCommentedHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastCommentedHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastCommentedHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:CommentedHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:CommentedHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:CommentedHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:CommentedHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:CommentedHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:CommentedHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:CommentedHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:CommentedHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastBasic delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastBasic delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastBasic delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastBasic delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastBasic delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastBasic delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastBasic delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastBasic delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:Basic delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:Basic delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:Basic delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:Basic delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:Basic delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:Basic delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:Basic delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:Basic delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastNoHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastNoHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastNoHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastNoHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastNoHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastNoHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:FastNoHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:FastNoHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:NoHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:NoHeader delimiter: '|' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:NoHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:NoHeader delimiter: ',' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:NoHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:NoHeader delimiter: ' ' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:NoHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: '"' strict_names: True
Reader:NoHeader delimiter: '\\s' fast_reader: {'enable': True} fill_values: [('', '0')] quotechar: "'" strict_names: True
Reader:Basic fast_reader: {'enable': True} fill_values: [('', '0')]

************************************************************************
** ERROR: Unable to guess table format with the guesses listed above. **
**                                                                    **
** To figure out why the table did not read, use guess=False and      **
** fast_reader=False, along with any appropriate arguments to read(). **
** In particular specify the format and any known attributes like the **
** delimiter.                                                         **
************************************************************************

```